File: saas-whatsapp-erp/frontend/backoffice/src/types/customer.ts
export interface Customer {
    id: string;
    companyId: string;
    name: string; // Single name field from Backend
    firstName?: string; // UI Compat
    lastName?: string; // UI Compat
    phone: string;
    currentState?: string; // CommercialState Enum
    taxId?: string;
    email?: string;
    address?: string;
    city?: string;
    country?: string;
    isActive?: boolean;
    whatsappConsent: boolean; // Added for MVP
    createdAt?: string;
    updatedAt?: string;
}

export interface CreateCustomerRequest {
    name: string;
    phone: string;
    email?: string;
    taxId?: string;
    address?: string;
    whatsappConsent: boolean;
}

export interface UpdateCustomerRequest {
    name?: string;
    phone?: string;
    email?: string;
    taxId?: string;
    address?: string;
    city?: string;
    notes?: string;
    whatsappConsent?: boolean;
}

File: saas-whatsapp-erp/backend/src/SaaS.Domain/Documents/Customer.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using SaaS.Domain.Enums;

namespace SaaS.Domain.Documents;

/// <summary>
/// Cliente - Persona o empresa que compra
/// Multi-tenant: Cada cliente pertenece a una Company
/// </summary>
public class Customer
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = string.Empty;

    /// <summary>
    /// ID de la empresa a la que pertenece este cliente (Multi-tenant)
    /// </summary>
    [BsonElement("companyId")]
    [BsonRepresentation(BsonType.ObjectId)]
    public string CompanyId { get; set; } = string.Empty;

    [BsonElement("name")]
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Teléfono principal (WhatsApp)
    /// </summary>
    [BsonElement("phone")]
    public string Phone { get; set; } = string.Empty;

    /// <summary>
    /// Estado comercial actual del cliente
    /// </summary>
    [BsonElement("currentState")]
    [BsonRepresentation(BsonType.String)]
    public CommercialState CurrentState { get; set; } = CommercialState.LEAD;

    /// <summary>
    /// Datos fiscales opcionales
    /// </summary>
    [BsonElement("taxId")]
    public string? TaxId { get; set; }

    [BsonElement("email")]
    public string? Email { get; set; }

    [BsonElement("address")]
    public string? Address { get; set; }

    [BsonElement("isActive")]
    public bool IsActive { get; set; } = true;

    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [BsonElement("updatedAt")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
